app/controllers/ReportingController.scala

```diff
@@ -272,4 +273,27 @@ trait ReportingController extends BaseController {
       Ok(Json.toJson(list))
     }
   }
+
+  def numericTestExtractReport(): Action[AnyContent] = Action.async { implicit request =>
+
+    val numericTestSchemeIds = schemeRepo.schemes.filter(scheme => scheme.siftEvaluationRequired &&
+      scheme.siftRequirement.contains(SiftRequirement.NUMERIC_TEST)).map(_.id)
+
+    val reports =
+      for {
+        applications <- reportingRepository.numericTestExtractReport().map(_.filter { app =>
+          val successfulSchemesSoFarIds = app.currentSchemeStatus.filter(_.result == Green.toString).map(_.schemeId)
+          successfulSchemesSoFarIds.exists(numericTestSchemeIds.contains)
+        })
+        contactDetails <- contactDetailsRepository.findByUserIds(applications.map(_.userId)).map(_.map(x => x.userId -> x)(breakOut).toMap)
```

Comment:
specifying the type on the val assignment would remove the need for the explicit toMap call